src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml.cs

```diff
@@ -29,28 +27,13 @@ public SearchViewModel ViewModel
             get { return DataContext as SearchViewModel; }
         }
 
-        private WorkspaceView workspaceView;
-        private DynamoView dynamoView;
-
         public InCanvasSearchControl()
         {
             InitializeComponent();
-
-            Loaded += (sender, e) =>
+            if (Application.Current != null)
             {
-                if (workspaceView == null)
-                {
-                    workspaceView = WpfUtilities.FindUpVisualTree<WorkspaceView>(Parent);
-                }
-
-                if (dynamoView != null) return;
-
-                dynamoView = WpfUtilities.FindUpVisualTree<DynamoView>(Parent);
-                if (dynamoView != null)
-                {
-                    dynamoView.Deactivated += (s, args) => { OnRequestShowInCanvasSearch(ShowHideFlags.Hide); };
-                }
-            };
+                Application.Current.Deactivated += (s, args) => { OnRequestShowInCanvasSearch(ShowHideFlags.Hide); };
+            }
```

Comment:
No, it is not related to click issue.
I made this change because `InCanvasSearchControl` control is specified inside template of context menu and that is why `WorkspaceView` and `DynamoView` are not found and subscribing to `Window.Deactivated` has not been done for `InCanvasSearchControl` which is in context menu. `ContextMenu` control hides automatically when window deactivates. But `Popup` control doesn't and that's why it needs this functionality to work.